modin/core/dataframe/base/partitioning/axis_partition.py

```diff
@@ -95,9 +96,11 @@ def _wrap_partitions(self, partitions):
         list
             List of wrapped remote partition objects.
         """
-        return [self.partition_type(obj) for obj in partitions]
+        return [self.partition_type(obj) for obj in partitions]  # type: ignore
```

Comment:
Why do we need the ignore?

Reply:
@shalearkane thank you, that makes sense. I suggest that we assert within this function that `instance_type` and `partition_type` are not `None`, because subclasses of `BaseDataframeAxisPartition` should set those fields to a non-`None` value.